File: backend/src/main/java/isis/projet/backend/entity/Etudiant.java
package isis.projet.backend.entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * Entité représentant un étudiant.
 * Un étudiant peut participer à plusieurs actions via Participe.
 */
@Entity
@Table(name = "etudiant")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Etudiant {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_etudiant")
    private Integer idEtudiant;

    @Column(name = "nom")
    private String nom;

    @Column(name = "prenom")
    private String prenom;

    @Column(name = "email", nullable = true)
    private String email;

    @Column(name = "promotion")
    private String promotion;

    /**
     * Un étudiant peut avoir plusieurs participations.
     * Relation "managed" côté étudiant.
     */
    @OneToMany(mappedBy = "etudiant", cascade = CascadeType.ALL)
    @JsonManagedReference(value = "etudiant-participations")
    private List<Participe> participations;
}

File: frontend/src/classe/Etudiant.js
export class Etudiant {
  constructor(idEtudiant, nom, prenom, email, promotion) {
    this._idEtudiant = idEtudiant;
    this._nom = nom;
    this._prenom = prenom;
    this._email = email;
    this._promotion = promotion;
  }

  // Getters and Setters
  get idEtudiant() { return this._idEtudiant; }
  set idEtudiant(value) { this._idEtudiant = value; }

  get nom() { return this._nom; }
  set nom(value) { this._nom = value; }

  get prenom() { return this._prenom; }
  set prenom(value) { this._prenom = value; }

  get email() { return this._email; }
  set email(value) { this._email = value; }

  get promotion() { return this._promotion; }
  set promotion(value) { this._promotion = value; }
}
